x/epochs/abci.go

```diff
@@ -29,11 +29,13 @@ func BeginBlocker(ctx sdk.Context, k keeper.Keeper) {
 			epochStart = true
 			epochInfo.CurrentEpoch = epochInfo.CurrentEpoch + 1
 		}
-
 		if epochStart {
 			epochInfo.CurrentEpochEnded = false
-			epochInfo.CurrentEpochStartTime = ctx.BlockTime()
-
+			if epochInfo.CurrentEpochStartTime.Equal(time.Time{}) { //genesis state, initialize
```

Comment:
If `CurrentEpochStartTime` initialization is done is genesis.go, there would be no way to figure out whether `CurrentEpoch` is the first, initializing epoch or not. 
Knowing whether `currentEpoch` is 0 or not within BeginBlocker is crucial in the sense that there is an exception for Epoch counting logic in the beginning. When an epoch is starting initially, `currentEpoch` is 0, while it gets incremented right away, creating an exceptional case to think in the Epoch logic. That is, at currentEpoch 0,it would create an error when trying to process with the same logic as any other currentEpoch, being needed to be processed exceptionally.